plugin-server/src/worker/vm/upgrades/historical-export/export-historical-events-v2.ts

```diff
@@ -445,6 +449,15 @@ export function addHistoricalEventsExportCapabilityV2(
             return
         }
 
+        // We bump the statusTime every minute to let the coordinator know we are still
+        // alive and we don't need to be resumed.
+        const interval = setInterval(meta.storage.set, 1000 * 60, payload.statusKey, {
+            ...payload,
+            done: false,
+            progress: progress,
+            statusTime: Date.now(),
+        } as ExportChunkStatus)
```

Comment:
I believe you'd be evaluating the arguments once and they would therefore be static. That can be fixed with something like:

```suggestion
        const interval = setInterval(() => meta.storage.set(1000 * 60, payload.statusKey, {
            ...payload,
            done: false,
            progress: progress,
            statusTime: Date.now(),
        } as ExportChunkStatus))
```

Can we add something to the tests that would catch this if it isn't already.

Reply:
Oh, thanks. Nice call. Any idea how to test this? I'm struggling with adding a mock delay to the function and making time pass so that we can see these calls.

Reply:
Jest can mock things like `setTimeout` and `setInterval` for you, you can try checking the examples on https://gist.github.com/apieceofbart/e6dea8d884d29cf88cdb54ef14ddbcc4 (random google search might not actually be any good)

Reply:
Yeah, I understand those examples, problem is that we are mocking `exportEvents` so we are returning from that immediately and advancing the function which leads to the `clearInterval` call, so we don't see the calls happen.

Reply:
Yeah, exactly :sweat_smile: I'm looking into it...